### PR TITLE
Feature/email notifications

### LINKED ISSUE #198: Send email notifications for reservations

### PR DESCRIPTION
**Pull request by KiOui**
Added the following email notifications:

- Email notifications to the admin email address when a (borrel) reservation is created
- Email notifications to the user when the status of a (borrel) reservation changes

Closes #198

### CODE REVIEW

File: website/borrel/services.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        settings.EMAIL_HOST_USER,
+        [borrel_reservation.user_created.email],
+    )
+    msg.attach_alternative(html_content, "text/html")
+
+    try:
+        return msg.send()
+    except SMTPException as e:
+        logger.error(e)
+        return False
```
Comment (JobDoesburg) [inline website/borrel/services.py:69]: Could you turn this into a separate `send_email()` helper function (for example in `tosti.services` `tosti.utils` or some other file?

File: website/borrel/signals.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+from django.dispatch import receiver
+
+from borrel.models import BorrelReservation
+from borrel.services import send_borrel_reservation_status_change_email
+
+
+@receiver(pre_save, sender=BorrelReservation)
+def send_borrel_reservation_status_change_email_receiver(sender, instance, **kwargs):
+    """Send a status change email when the status of a BorrelReservation has been changed."""
+    if instance.pk is not None and instance.user_created is not None:
```
Comment (JobDoesburg) [inline website/borrel/signals.py:11]: Better code quality
```suggestion
    if instance.pk is None:
        return
```

File: website/borrel/services.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    html_content = template.render(context)
+    text_content = template_text.render(context)
+
+    msg = EmailMultiAlternatives(
+        "TOSTI: Borrel Reservation status change",
+        text_content,
+        settings.EMAIL_HOST_USER,
+        [borrel_reservation.user_created.email],
```
Comment (JobDoesburg) [inline website/borrel/services.py:61]: Maybe the association should also get this email in the CC? Maybe add a notification email field to associations?

File: website/borrel/templates/email/borrel_reservation.txt [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+Title reservation:
+{{ borrel_reservation.title }}
+Start:
+{{ borrel_reservation.start }}
+{% if borrel_reservation.end %}End:
+{{ borrel_reservation.end }}{% endif %}
+{% if borrel_reservation.comments %}Comments:
+{{ borrel_reservation.comments }}{% endif %}
+
+To view details, accept or reject this reservation, visit https://{{ domain }}{% url 'admin:borrel_borrelreservation_change' object_id=borrel_reservation.id %}.
```
Comment (JobDoesburg) [inline website/borrel/templates/email/borrel_reservation.txt:18]: Can we automatically extract this from the html perhaps? I already see people updating the HTML template but forgetting to update the TXT template.

**Review by JobDoesburg — CHANGES_REQUESTED** [review]

File: website/tosti/settings/base.py [hunk 5]
```diff
@@ -171,8 +172,13 @@
     'TANTALUS_ENDPOINT_URL': ('', 'Endpoint for Tantalus integration', str),
     'TANTALUS_USERNAME': ('', 'Username for Tantalus integration', str),
     'TANTALUS_PASSWORD': ('', 'Password for Tantalus integration', str),
+    'SEND_EMAIL_TO': ('noreply@example.com', 'Where to send e-mail notifications to', str),
```
[the remark this answers is not in the document]
Comment (KiOui) [reply website/tosti/settings/base.py:175]: This is the e-mail address where notifications for creation of reservation requests go to

File: website/borrel/services.py (hunk 3 above)
Comment (KiOui) [reply website/borrel/services.py:61]: Let's do this another time.

File: website/tosti/settings/base.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (JobDoesburg) [reply website/tosti/settings/base.py:175]: Then maybe make this a little more descriptive

File: website/borrel/templates/email/borrel_reservation.txt (hunk 4 above)
Comment (KiOui) [reply website/borrel/templates/email/borrel_reservation.txt]: Yes we can

File: website/venues/services.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+from constance import config
+from django.contrib.sites.models import Site
+from django.template.loader import get_template
+
+from tosti.services import send_email
+from venues import models
+
+
+logger = logging.getLogger(__name__)
```
Comment (JobDoesburg) [inline website/venues/services.py:11]: this is not required anymore now

File: website/borrel/services.py [hunk 7]
```diff
@@ -0,0 +1,38 @@
+import logging
+
+from constance import config
+from django.contrib.sites.models import Site
+from django.template.loader import get_template
+
+from borrel import models
+from tosti.services import send_email
+
+logger = logging.getLogger(__name__)
```
Comment (JobDoesburg) [inline website/borrel/services.py:10]: This is not required anymore

**Review by JobDoesburg — APPROVED** [review]

**Pull request merged**